Sources/Bookmarks/BookmarkListViewModel.swift

```diff
@@ -120,21 +120,26 @@ public class BookmarkListViewModel: BookmarkListInteracting, ObservableObject {
             return
         }
 
-        guard let children = parentFolder.children,
-              fromIndex < children.count,
-              toIndex < children.count else {
+        let visibleChildren = parentFolder.childrenArray
+
+        guard fromIndex < visibleChildren.count,
+              toIndex < visibleChildren.count else {
             errorEvents?.fire(.indexOutOfRange(.bookmarks))
             return
         }
 
-        guard let actualBookmark = children[fromIndex] as? BookmarkEntity,
-              actualBookmark == bookmark else {
+        guard visibleChildren[fromIndex] == bookmark else {
             errorEvents?.fire(.bookmarksListIndexNotMatchingBookmark)
             return
         }
 
+        // Take into account bookmarks that are pending deletion
         let mutableChildrenSet = parentFolder.mutableOrderedSetValue(forKeyPath: #keyPath(BookmarkEntity.children))
-        mutableChildrenSet.moveObjects(at: IndexSet(integer: fromIndex), to: toIndex)
+
+        let actualFromIndex = mutableChildrenSet.index(of: bookmark)
+        let actualToIndex = mutableChildrenSet.index(of: visibleChildren[toIndex])
```

Comment:
These calls may return `NSNotFound` (which I believe equals -1), but `IndexSet` initializer fails when initialized with a negative index. I realize it's unlikely to happen, but perhaps it's still worth to guard against it?

Reply:
As `visibleChildren` call above already fetched `children` relationship, this one should not return different (smaller) result set, and both bounds and existence if bookmark has been validated earlier. I'd be surprised to see this fail and I wanted to avoid having to guard and report another pixel here.

Reply:
Sure, perhaps an assertion makes sense to make intentions clear?

Reply:
I've added assertion, and applied same patter to favorites.

Reply:
Awesome, thanks!